### PR TITLE
docs(README): fix typos

### PR DESCRIPTION
**Pull request by Mnigos**
Fixing simple typos inside `README.md`

Also I'm suggesting to embed a gif that really uses `nestjs-trpc` and not vanilla trpc server.

### CODE REVIEW

**Comment by KevinEdry** [comment]
I am painfully aware that this readme isn't in the best shape 😅, I've mentioned it in the original trpc issue. Would you mind adding the gif to this PR?

**Comment by Mnigos** [comment]
> I am painfully aware that this readme isn't in the best shape 😅, I have spent too much time on the docs and neglected this readme, I think I also mentioned it in the original trpc issue. Would you mind adding the gif to this PR?

Sure!

**Comment by Mnigos** [comment]
@KevinEdry I Edited code formatting a little bit let me know what you are thinking, but I'm not sure what `@Procedure` decorator and `ProtectedProcedure` are doing. Is it like a guard for tRPC? I don't see it covered anywhere in documentation.

File: README.md [hunk 1]
```diff
@@ -33,27 +35,30 @@
 ## Quickstart
 
 ### Installation
+
 To install **NestJS tRPC** with your preferred package manager, you can use any of the following commands:
 
 ```shell
 # npm
-npm install trpc-nestjs
+npm install trpc-nestjs zod
```
Comment (KevinEdry) [inline README.md:43]: We should also add the `@trpc/server` here since both `zod` and `@trpc/server` are peer dependencies.

File: README.md [hunk 2]
```diff
@@ -63,18 +68,16 @@ const userSchema = z.object({
 @Router()
 class UserRouter {
   constructor(
-    @Inject(UserService) private readonly userService: UserService
```
Comment (KevinEdry) [inline README.md:66]: I'd rather move this to the top and keep the Idiomatic way of the `@Inject()` decorator to emphasis that DI works seamlessly with this library.

File: README.md [hunk 3]
```diff
@@ -63,18 +68,16 @@ const userSchema = z.object({
...
   @Query({ output: z.array(userSchema) })
   async getUsers() {
     try {
-      const users = await this.userService.getUsers();
-      return users;
-    } catch (e: unknown) {
-      throw new TRPCError("An error has occured when trying to get users.", "INTERNAL_SERVER_ERROR", e)
+      return this.userService.getUsers();
+    } catch (error: unknown) {
+      throw new TRPCError("An error has occured when trying to get users.", "INTERNAL_SERVER_ERROR", error)
```
Comment (KevinEdry) [inline README.md:80]: In general, the way I handle errors here isn't correct, see https://trpc.io/docs/v10/server/error-handling
I need to pass an object rather then single parameters here.

**Review by KevinEdry — CHANGES_REQUESTED** [review]
> @KevinEdry I Edited code formatting a little bit let me know what are you thinking, but I'm not sure what `@Procedure` decorator and `ProtectedProcedure` are doing. Is it like a guard for tRPC? I don't see it covered anywhere in documentation.

The `@Procedure` is the previous implementation of [Middlewares](https://www.nestjs-trpc.io/docs/middlewares). The docs are updated with the correct examples.

Looks good overall, there are still a couple of things that needs to be corrected.
Thanks for taking the time to do this!

**Comment by Mnigos** [comment]
@KevinEdry I've applied your requested changes and removed some deprecated code from husky hook.

**Comment by Mnigos** [comment]
> Looks awesome, thanks for helping with the README file! I wish there was also a gif using the library, but it's not a must-have.

I will try to dedicate some of my time to create this gif

**Pull request merged**